File: CarHomeWidget/src/main/java/com/anod/car/home/prefs/model/AppTheme.java
package com.anod.car.home.prefs.model;

import com.anod.car.home.R;

/**
 * @author algavris
 * @date 09/04/2016.
 */
public class AppTheme {
    public static final int THEME_GRAY = 0;
    public static final int THEME_HOLO = 1;

    public static int getNameResource(int theme) {
        return (theme == THEME_GRAY) ? R.string.theme_gray : R.string.theme_holo;
    }

    public static int getMainResource(int theme) {
        return (theme == THEME_GRAY) ? R.style.AppThemeGray : R.style.AppThemeHolo;
    }

    public static int getTransparentResource(int theme) {
        return (theme == THEME_GRAY) ? R.style.AppThemeGray_Transparent
                : R.style.AppThemeHolo_Transparent;
    }

    public static int getNoActionBarResource(int theme) {
        return (theme == THEME_GRAY) ? R.style.AppThemeGray_NoActionBar
                : R.style.AppThemeHolo_NoActionBar;
    }

    public static int getMain() {
        return R.style.AppThemeGray;
    }

    public static int getBackgroundResource(int theme) {
        return (theme == THEME_GRAY) ? R.drawable.panel_item_bg_grey
                : R.drawable.panel_item_bg_dark;
    }
}

File: CarHomeWidget/src/main/java/com/anod/car/home/prefs/model/AppTheme.kt
package com.anod.car.home.prefs.model

import androidx.annotation.ColorRes
import androidx.annotation.StyleRes
import com.anod.car.home.R

/**
 * @author algavris
 * @date 09/04/2016.
 */
class AppTheme(val themeIdx: Int) {

    companion object {
        const val gray = 0
        const val dark = 1
    }

    val main: Int
        @StyleRes get() = R.style.AppThemeGray

    val mainResource: Int
        @StyleRes get() = if (themeIdx == gray) R.style.AppThemeGray else R.style.AppThemeHolo

    val transparentResource: Int
        @StyleRes get() = if (themeIdx == gray) R.style.AppThemeGray_Transparent else R.style.AppThemeHolo_Transparent

    val noActionBarResource: Int
        @StyleRes get() = if (themeIdx == gray) R.style.AppThemeGray_NoActionBar else R.style.AppThemeHolo_NoActionBar

    val backgroundResource: Int
        @ColorRes get() =  if (themeIdx == gray) R.color.panel_background_grey else R.color.panel_background_dark

    val alert: Int
        @StyleRes get() = if (themeIdx == gray) R.style.AlertGray else R.style.AlertHolo

    val dialog: Int
        @StyleRes get() = if (themeIdx == gray) R.style.DialogGrey else R.style.DialogHolo

}
